lib/modules.nix

```diff
@@ -267,7 +256,7 @@ rec {
 
      'opts' is a list of modules.  Each module has an options attribute which
      correspond to the definition of 'loc' in 'opt.file'. */
-  mergeOptionDecls = loc: opts:
+  mergeOptionDecls = loc: opts: defns:
```

Comment:
This is more of a comment for thought, not a problem of itself.

The comment above says "In general, there should be only one declaration of each option.". Which makes me think - why do we have to "merge" options declarations if `length opts == 1`? Isn't there a shortcut for this case?

At least, if I comment out `bothHave` checks, I have little speedup.

Reply:
Hm yeah, I tried out adding a special case for that, but it made the code rather verbose, so much that I don't think it's worth it. Maybe that could be done better though.